From: Miklos Szeredi <miklos@szeredi.hu>
Subject: [FUSE] Add the offset field to fuse_dirent
Patch-mainline: 

This patch adds an offset field to fuse_dirent structure.  This will give
userspace filesystems more flexibility in implementing a readdir method.

Signed-off-by: Miklos Szeredi <miklos@szeredi.hu>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Acked-by: Takashi Iwai <tiwai@suse.de>

---

 25-akpm/fs/fuse/dir.c        |    2 +-
 25-akpm/include/linux/fuse.h |    1 +
 2 files changed, 2 insertions(+), 1 deletion(-)

diff -puN fs/fuse/dir.c~fuse-read-only-operations-add-offset-to-fuse_dirent fs/fuse/dir.c
--- 25/fs/fuse/dir.c~fuse-read-only-operations-add-offset-to-fuse_dirent	Thu Mar 31 15:04:39 2005
+++ 25-akpm/fs/fuse/dir.c	Thu Mar 31 15:04:39 2005
@@ -197,7 +197,7 @@ static int parse_dirfile(char *buf, size
 			break;
 
 		over = filldir(dstbuf, dirent->name, dirent->namelen,
-			       file->f_pos, dirent->ino, dirent->type);
+			       dirent->off, dirent->ino, dirent->type);
 		if (over)
 			break;
 
diff -puN include/linux/fuse.h~fuse-read-only-operations-add-offset-to-fuse_dirent include/linux/fuse.h
--- 25/include/linux/fuse.h~fuse-read-only-operations-add-offset-to-fuse_dirent	Thu Mar 31 15:04:39 2005
+++ 25-akpm/include/linux/fuse.h	Thu Mar 31 15:04:39 2005
@@ -120,6 +120,7 @@ struct fuse_out_header {
 
 struct fuse_dirent {
 	__u64	ino;
+	__u64	off;
 	__u32	namelen;
 	__u32	type;
 	char name[0];
_
